syntax = "proto3";

message Simple {
  string str = 1;
  int64 i64 = 2;
  uint64 ui64 = 3;
  bool  b = 4;
}

message Complex {
  Simple simple = 1;
  string opt_str = 2;
  Simple opt_simple = 3;
}